init PR templates (#23726)
gravitational/teleport: .github/docs_pr_change_template.md

## Description

<!-- Describe the changes made in this PR -->

## Checklist

> - [ ] This change matches the [docs style guide](https://goteleport.com/docs/contributing/documentation/style-guide/), and the terms used in this change correspond to those in [Teleport Core Concepts](https://goteleport.com/docs/core-concepts)

Reviewer: Based on my conversations with Sasha and Xin, this seems like more of the responsibility of the docs team. I don't think it hurts to capture it here (engineering should make an effort) but we don't have the wholistic view that you do.

Author: We've also talked about the docs team reviewing all docs PRs, and were told that was unrealistic. It can't be our sole responsibility if we're also not reviewing every docs change.

> - [ ] I viewed the changes on the dev server (`yarn dev`) to confirm they render without error

Reviewer: Are there good instructions on how to do this? 

Confession, I've merged dozens (if not more) docs PRs and I've never done this 😇

Author: How about:

```suggestion
- [ ] I viewed the changes on the [dev server](https://github.com/gravitational/docs/blob/main/README.md#getting-started) to confirm they render without error
```

Reviewer: Perfect!

## Description

<!-- Describe the changes made in this PR -->

## Checklist
- [ ] This change matches the [docs style guide](https://goteleport.com/docs/contributing/documentation/style-guide/)

> - [ ] I viewed the changes on the sev server (`yarn dev`) to confirm they render without error

Reviewer 2: ```suggestion
- [ ] I viewed the changes on the dev server (`yarn dev`) to confirm they render without error
```

> - [ ] I viewed the changes on the dev server (`yarn dev`) to confirm they render without error

Reviewer 2: We could probably condense these by asking the reader to look at `/docs/contributing/documentation`

Reviewer 2: Initially, I was thinking of a way to use a reference to the contribution guide to consolidate this and the previous item. However, I also think it would make sense to emphasize that the reader should view the changes on the dev server.

Do you see another other ways we can reduce the length of the checklist? Without the spell checking and Core concepts items, the checklist for docs changes has four items (which I think is okay) and the list for new docs pages has seven. Is there a way to trim the list for new docs pages to five items?

Author: I've done some additional consolidation.

> - [ ] The terms used in this change correspond to those in [Teleport Core Concepts](https://goteleport.com/docs/core-concepts)

Reviewer 2: It would be nice to be able to lint for this one using gravitational/docs#238, though we still might need to keep this list item here. What do you think?

Author: I think if that PR gets merged we should definitely remove this line.

> - [ ] I spellchecked the changes

Reviewer 2: Hopefully we won't need this if we merge this PR: https://github.com/gravitational/teleport/pull/23783

> - [ ] I referenced any issues this PR will close

Reviewer 2: If we wanted to make this shorter, we could also remove this item, since it doesn't affect the docs as much as our own internal processes (i.e., it would take some extra time to close outdated issues)

Author: Sure.

```suggestion
```

- [ ] I checked for existing PRs related to this change.

> - [ ] I added appropriate backport labels (we support the previous two versions of Teleport)

Reviewer 2: ```suggestion
- [ ] I added appropriate backport labels (we support the previous two major versions of Teleport)
```
